### PR TITLE
Fixed credit card type number validation error

### PR DESCRIPTION
**Pull request by emirajbbd**
### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by dmytro-ch** [comment]
Hi @emirajbbd,
could you please provide information about the issue and steps to reproduce?
Also, some descriptions for the reason of issue and applied solution are welcomed.
Thank you!

**Comment by emirajbbd** [comment]
**Description:**
When we have PayPal Payments Pro/PayPal Payflow Pro enabled to process Debit or Credit Card and user tries to place an order using his credit card info, it always sends validation error "Please enter a valid credit card type number" though the card was valid.

**Steps to reproduce:**
1. Enable PayPal Payments Pro/PayPal Payflow Pro along with other  Credit Card processor like Authorize.net/Merchantesolutions/module-gateway(disabled)  on the backend
2. Go to checkout payment step
3. Add correct Credit/Debit card info
4. Error message display "Please enter a valid credit card type number"

File: app/code/Magento/Payment/view/base/web/js/model/credit-card-validation/validator.js [hunk 1]
```diff
@@ -35,7 +36,13 @@ define([
                 cardInfo = creditCardNumberValidator(number).card;
 
                 for (i = 0, l = allowedTypes.length; i < l; i++) {
-                    if (cardInfo.title == allowedTypes[i].type) { //eslint-disable-line eqeqeq
+                    
+                    type = allowedTypes[i].type;
+                    
+                     if( !!type && type.constructor === Array && type.includes(cardInfo.title)){ // Check if an arry
```
Comment (dmytro-ch) [inline app/code/Magento/Payment/view/base/web/js/model/credit-card-validation/validator.js:42]:
```suggestion
                    if (!!type && type.constructor === Array && type.includes(cardInfo.title)) { // Check if an array
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
Hi @emirajbbd , could you please check and fix the static test failures?
Once the changes are applied, please squash all commits into a single one in order to cleanup history.
Thank you!

**Comment by dmytro-ch** [comment]
Hi, @emirajbbd, I'm Closing this PR now due to inactivity.
Feel free to reopen and update if you wish to continue.
Thank you!

**Pull request closed without merging**